core/evalir.ml

```diff
@@ -638,11 +639,11 @@ struct
             ~predicate
           |> Lens_errors.unpack_sort_select_result ~die:(eval_error "%s")
         in
-        apply_cont cont env (`Lens (Value.LensSelect {lens; predicate; sort}))
+        apply_cont cont env (`Lens (db, Value.LensSelect {lens; predicate; sort}))
     | LensJoin { left; right; on; del_left; del_right; _ } ->
         let open Lens in
-        value env left >|= get_lens >>= fun lens1 ->
-        value env right >|= get_lens >>= fun lens2 ->
+        value env left >|= unpack_lens >>= fun (db1, lens1) ->
+        value env right >|= unpack_lens >>= fun (_db2, lens2) ->
```

Comment:
Should it be allowed to build a join lens where the underlying databases are different?

Reply:
I've just committed a check for this. It used to happen in the old code.